### DISCUSSION TITLE
Close checkout window confirmation

### DISCUSSION THREAD

**Discussion by jaonoctus** [btcpayserver/btcpayserver#2529] [Technical Support] [answered]
I would like to modify the checkout experience on my btcpayserver:

> When the customer tries to close the checkout window, he receives an alert in the browser if he really wants to close it.

I did some research and I think I could modify the file [CheckoutNoScript.cshtml](https://github.com/btcpayserver/btcpayserver/blob/0eace936b0882f3c53772cbd56b9c9989731e805/BTCPayServer/Views/Invoice/CheckoutNoScript.cshtml), adding something like

```html
<script>
      window.onbeforeunload = function (e) {
        return "You haven't paid the invoice yet. Do you really want to leave?";
      }
</script>
```

The thing is, i'm running the [btcpayserver-docker](https://github.com/btcpayserver/btcpayserver-docker). How can I make these changes work?

**Comment by bolatovumar** [comment, marked as answer]
@jaonoctus you can't modify `*.cshtml` files like this unless you build your own version of BTCPay from source and run it.

**Reply by jaonoctus** [reply]
@bolatovumar thanks for your reply!

I could fork the project and do it then.

But do you have any docs or steps to how to build it?

**Reply by Zaxounette** [reply]
The following documentation explaines how to fork and build your branch to test a new version of BTCPay before hand.
https://docs.btcpayserver.org/Contribute/ContributeDev/ContributeDevTest/#using-docker-images-for-mainnet-testing

To maintain your own fork of BTCPay through time, it's the same process, just replace the default github branch link with your own where necessary.
